plugins/org.jkiss.dbeaver.model.jdbc/src/org/jkiss/dbeaver/model/impl/jdbc/struct/JDBCTable.java

```diff
@@ -148,8 +148,10 @@ public DBCStatistics readData(
         // Some criteria doesn't work without alias
         // (e.g. structured attributes in Oracle or composite types in PostgreSQL requires table alias)
         String tableAlias = null;
+        String tableFullName = getTableName();
         if (needAliasInSelect(dataFilter, rowIdAttribute, dataSource)) {
-            tableAlias = DEFAULT_TABLE_ALIAS;
+            String[] parts = tableFullName.split("\\.");
```

Comment:
Better use dialect.getStructSeparator() for dot. I have never seen a database where it's not a dot, but we have this in SQL Dialect (maybe for some reason)